# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/experimental/JSStencil.h
# Commit: 2bc5949902ac
# Full Hash: 2bc5949902acc042205fc7e85260bbbc37dbedc7
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2021-09-18 09:46:57
# Regressor Bug: 1688788
# File Overlap Count: 1
# Description:
#   Bug 1688788 - Convert ScriptPreloader to manage Stencils, not Scripts. r=kmag,tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113013
# ==============================================================================

diff -r e660e2c3c3d1 -r 2bc5949902ac js/public/experimental/JSStencil.h
--- a/js/public/experimental/JSStencil.h	Sat Sep 18 01:29:45 2021 +0000
+++ b/js/public/experimental/JSStencil.h	Sat Sep 18 01:29:45 2021 +0000
@@ -14,8 +14,10 @@
  * building script loaders.
  */
 
-#include "mozilla/RefPtr.h"  // RefPtr, already_AddRefed
-#include "mozilla/Utf8.h"    // mozilla::Utf8Unit
+#include "mozilla/MemoryReporting.h"  // mozilla::MallocSizeOf
+#include "mozilla/RefPtr.h"           // RefPtr, already_AddRefed
+#include "mozilla/Utf8.h"             // mozilla::Utf8Unit
+#include "mozilla/Vector.h"           // mozilla::Vector
 
 #include <stddef.h>  // size_t
 
@@ -24,7 +26,7 @@
 #include "js/CompileOptions.h"              // JS::ReadOnlyCompileOptions
 #include "js/OffThreadScriptCompilation.h"  // JS::OffThreadCompileCallback
 #include "js/SourceText.h"                  // JS::SourceText
-#include "js/Transcoding.h"
+#include "js/Transcoding.h"                 // JS::TranscodeSource
 
 struct JS_PUBLIC_API JSContext;
 
@@ -81,6 +83,10 @@
 extern JS_PUBLIC_API JSScript* InstantiateGlobalStencil(
     JSContext* cx, const ReadOnlyCompileOptions& options, Stencil* stencil);
 
+// Return true if the stencil relies on external data as a result of XDR
+// decoding.
+extern JS_PUBLIC_API bool StencilIsBorrowed(Stencil* stencil);
+
 // Instantiate a module Stencil and return the associated object. Inside the
 // engine this is a js::ModuleObject.
 extern JS_PUBLIC_API JSObject* InstantiateModuleStencil(
@@ -96,6 +102,9 @@
 DecodeStencil(JSContext* cx, const ReadOnlyCompileOptions& options,
               const TranscodeRange& range, Stencil** stencilOut);
 
+extern JS_PUBLIC_API size_t SizeOfStencil(Stencil* stencil,
+                                          mozilla::MallocSizeOf mallocSizeOf);
+
 extern JS_PUBLIC_API OffThreadToken* CompileToStencilOffThread(
     JSContext* cx, const ReadOnlyCompileOptions& options,
     SourceText<char16_t>& srcBuf, OffThreadCompileCallback callback,
@@ -109,6 +118,15 @@
 extern JS_PUBLIC_API already_AddRefed<Stencil> FinishOffThreadCompileToStencil(
     JSContext* cx, OffThreadToken* token);
 
+extern JS_PUBLIC_API OffThreadToken* DecodeMultiOffThreadStencils(
+    JSContext* cx, const ReadOnlyCompileOptions& options,
+    mozilla::Vector<TranscodeSource>& sources,
+    OffThreadCompileCallback callback, void* callbackData);
+
+extern JS_PUBLIC_API bool FinishMultiOffThreadStencilDecoder(
+    JSContext* cx, OffThreadToken* token,
+    mozilla::Vector<RefPtr<Stencil>>* stencils);
+
 }  // namespace JS
 
 namespace mozilla {